tests/test_cluster_model.py

```diff
@@ -0,0 +1,141 @@
+import itertools
+import random
+
+import numpy as np
+import numpy.testing as npt
+import pandas as pd
+
+from collections import defaultdict
+from numpy.random import randint
+from random import SystemRandom
+from pandas.util.testing import assert_series_equal
+from sklearn.datasets import make_regression
+from sklearn.linear_model import LinearRegression
+from sklearn.model_selection import train_test_split
+
+from vf_portalytics.cluster_model import ClusterModel
+from vf_portalytics.model import PredictionModel
+
+
+def make_dataset(random_state, n_informative, column_names, **kwargs):
+    x, y = make_regression(
+
+        n_samples=1000,
+        n_features=5,
+        noise=0 if random_state == 1 else 10,
+        bias=10 if random_state == 1 else 1000,
+        n_informative=min(n_informative, 5),
+        random_state=random_state
+    )
+    cryptogen = SystemRandom()
+    x = pd.DataFrame(x)
+    x.columns = [name for name in column_names]
+    x = x.assign(**kwargs)
+    x['yearweek'] = randint(1, 54, len(x))
+    # pack_type (original_product_dimension_44) 0: 'Can', 1: 'Bottle'
+    x['original_product_dimension_44'] = [cryptogen.randrange(1) for i in range(len(x))]
+    x['original_product_dimension_47'] = [cryptogen.randrange(4) for i in range(len(x))]
+    x['original_product_dimension_48'] = [cryptogen.randrange(4) for i in range(len(x))]
+    x['original_product_dimension_42'] = x['original_product_dimension_48'].map(lambda x: x % 2)
+
+    return x, pd.Series(y)
+
+
+def make_dict():
+    """
+    Creates a dictionary with keys all the combinations between the weeks of the year and the pack types
+    In this case we have only one pack type, in order to check if when we dont have the pack type in the dict,
+    the model will predict 0.
+    """
+    all_list = [list(range(1, 54)), [0]]
+    keys = list(itertools.product(*all_list))
+    values = [random.choice(np.linspace(-2.5, 2.5, num=500)) for i in range(len(keys))]
+    return dict(zip(keys, values))
```

Comment:
You could simplify this function by using a dictionary comprehension instead. That way you can iterate directly over the `product` iterator.

```suggestion
    return { key: random.choice(np.linspace(-2.5, 2.5, num=500)) for key in itertools.product(*all_list) }
```